### ISSUE TITLE
Remove references to pub.dartlang.org

### ISSUE THREAD

**Issue by kwalrath** [dart-lang/site-www#1355]
Docs that used to be on dartdocs.org are now at pub.dartlang.org. We should replace mentions of dartdocs.org with references to the `pub-api` config variable, as appropriate. While we're at it, we should fix the few remaining direct references to pub.dartlang.org to use `pub`, `pub-api`, or `pub-pkg`.

```
$ fgrep -R dartdocs.org src
src/_articles/libraries/zones.md:  [Chain class](https://www.dartdocs.org/documentation/stack_trace/1.6.5/stack_trace/Chain-class.html)
src/_tutorials/server/httpserver.md:[VirtualDirectory]: https://www.dartdocs.org/documentation/http_server/latest/http_server/VirtualDirectory-class.html
src/_tutorials/server/cmdline.md:| <a href="https://www.dartdocs.org/documentation/args/latest/args/ArgParser-class.html" target="_blank">ArgParser</a> | A class that parses command-line arguments |
src/_tutorials/server/cmdline.md:| <a href="https://www.dartdocs.org/documentation/args/latest/args/ArgResults-class.html" target="_blank">ArgResults</a> | The result of parsing command-line arguments using ArgParser. |
src/_tutorials/server/cmdline.md:<a href="https://www.dartdocs.org/documentation/args/latest/index.html" target="_blank">API docs</a>
src/_tutorials/server/cmdline.md:| <a href="https://www.dartdocs.org/documentation/args/latest/args/ArgParser-class.html" target="_blank">ArgParser</a> | A class that transforms a list of raw arguments and into a set of options, flags, and remaining values. |
src/_tutorials/server/cmdline.md:| <a href="https://www.dartdocs.org/documentation/args/latest/args/ArgResults-class.html" target="_blank">ArgResults</a> | The result of parsing raw command line arguments using ArgParser. |
src/_tutorials/server/cmdline.md:<a href="https://www.dartdocs.org/documentation/args/latest/index.html" target="_blank">args</a>
src/_guides/libraries/library-tour.md:[dartdocs.org]: https://www.dartdocs.org/
src/community/index.md:* [dartdocs.org](http://www.dartdocs.org) - Documentation for pub packages
src/community/who-uses-dart.md:[dartdocs.org](http://www.dartdocs.org)
```

**Comment by kwalrath** [comment]
Fixed by #1357.

**Comment by kwalrath** [comment]
Except for the references to pub.dartlang.org...

```
$ fgrep -R pub.dartlang.org src
src/_includes/page-footer.html:            <li><a href="https://pub.dartlang.org/">Pub packages</a></li>
src/index.html:              <h4><a href="https://pub.dartlang.org/">Dart packages</a></h4>
src/index.html:            <a href="https://pub.dartlang.org/">ecosystem</a> of
src/_guides/language/analysis-options.md:[invalid_assignment]: https://pub.dartlang.org/documentation/analyzer/latest/analyzer/StaticTypeWarningCode/INVALID_ASSIGNMENT-constant.html
```
